tiledb/sm/query/readers/sparse_global_order_reader.cc

```diff
@@ -500,22 +494,58 @@ bool SparseGlobalOrderReader<BitmapType>::create_result_tiles() {
   }
 
   read_state_.done_adding_result_tiles_ = done_adding_result_tiles;
-  return tiles_found;
+
+  // Return the list of tiles added.
+  std::vector<ResultTile*> created_tiles;
+  for (uint64_t i = 0; i < result_tiles.size(); i++) {
+    TileListIt it = result_tiles[i].begin();
+    std::advance(it, rt_list_num_tiles[i]);
+    for (; it != result_tiles[i].end(); ++it) {
+      created_tiles.emplace_back(&*it);
+    }
+  }
+
+  return created_tiles;
+}
+
+template <class BitmapType>
+void SparseGlobalOrderReader<BitmapType>::clean_tile_list(
+    std::vector<ResultTilesList>& result_tiles) {
+  // Clear result tiles that are not necessary anymore.
+  auto fragment_num = fragment_metadata_.size();
+  std::mutex ignored_tiles_mutex;
+  throw_if_not_ok(parallel_for(
+      storage_manager_->compute_tp(), 0, fragment_num, [&](uint64_t f) {
+        auto it = result_tiles[f].begin();
+        while (it != result_tiles[f].end()) {
+          if (it->result_num() == 0) {
+            {
+              std::unique_lock<std::mutex> lck(ignored_tiles_mutex);
```

Comment:
It seems a bit weird to be creating a mutex to coordinate access to a member variable. I assume this is just an instance of "I know this is not used by multiple threads anywhere else" that I don't have the context for. But might be worth adding a comment along the lines of why we're creating a mutex here instead of as a member variable that controls all access to `tmp_read_state_`.

Reply:
Yeah that was a sore spot for me as well when I wrote the code... I was worried to add it to add_ignored_tile as one call is in a performance sensitive path, but I realized it shouldn't be called often there so I moved the mutex to the class. Looks much better now!